File: box-server/box-domain/src/main/java/com/boxai/domain/tool/Tool.java
package com.boxai.domain.tool;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class Tool {

    private Long id;
    private Long workspaceId;
    private String name;
    private String toolKey;
    private String description;
    private String type;
    private String inputSchemaJson;
    private String outputSchemaJson;
    private Integer status;
    private Long createdBy;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: box-web/src/api/tool.ts
import http, { type Result } from './http'

export interface HttpToolConfigVO {
  method: string
  url: string
  headersJson?: string
  queryParamsJson?: string
  bodyType?: string
  bodyTemplate?: string
  timeoutMs?: number
  allowRedirect?: boolean
}

export interface ToolVO {
  id: number
  name: string
  toolKey: string
  description?: string
  type: string
  inputSchemaJson?: string
  outputSchemaJson?: string
  status: number
  httpConfig?: HttpToolConfigVO
  createdAt: string
  updatedAt: string
}

export function listTools() {
  return http.get<Result<ToolVO[]>>('/tools')
}

export function createTool(payload: {
  name: string
  toolKey: string
  description?: string
  type: string
  httpConfig?: Partial<HttpToolConfigVO>
}) {
  return http.post<Result<ToolVO>>('/tools', payload)
}

export function deleteTool(id: number) {
  return http.delete<Result<void>>(`/tools/${id}`)
}

export function testTool(id: number) {
  return http.post<Result<{ statusCode: number; body: string; durationMs: number }>>(`/tools/${id}/test`)
}
